web/management/commands/create_test_data.py

```
@@ -549,3 +582,254 @@
+                    max_score=100,
+                    completed=True,
+                    start_time=timezone.now() - timedelta(days=random.randint(1, 5)),
+                    end_time=timezone.now() - timedelta(days=random.randint(0, 4)),
+                    answers=json.dumps(self.generate_mock_answers(course_exam)),
+                )
+                self.stdout.write(f"Created submission for {student.username} - {course_exam.title}")
+        
+        # Create session exams
+        for session in sessions:
+            # Create session exam (50% probability)
+            if random.random() < 0.5:
```

Ruff (0.8.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)